File: tests/testdata/codegen/azure-sa-pp/nodejs/azure-sa.ts
import * as codeinfra from "@codeinfra/codeinfra";
import * as azure from "@codeinfra/azure";

const config = new codeinfra.Config();
// The name of the storage account
const storageAccountNameParam = config.require("storageAccountNameParam");
// The name of the resource group
const resourceGroupNameParam = config.require("resourceGroupNameParam");
const resourceGroupVar = azure.core.getResourceGroup({
    name: resourceGroupNameParam,
});
const locationParam = config.get("locationParam") || resourceGroupVar.then(resourceGroupVar => resourceGroupVar.location);
const storageAccountTierParam = config.get("storageAccountTierParam") || "Standard";
const storageAccountTypeReplicationParam = config.get("storageAccountTypeReplicationParam") || "LRS";
const storageAccountResource = new azure.storage.Account("storageAccountResource", {
    name: storageAccountNameParam,
    accountKind: "StorageV2",
    location: locationParam,
    resourceGroupName: resourceGroupNameParam,
    accountTier: storageAccountTierParam,
    accountReplicationType: storageAccountTypeReplicationParam,
});
export const storageAccountNameOut = storageAccountResource.name;

File: tests/testdata/codegen/azure-sa-pp/go/azure-sa.go
package main

import (
	"github.com/khulnasoft/codeinfra-azure/sdk/v4/go/azure/core"
	"github.com/khulnasoft/codeinfra-azure/sdk/v4/go/azure/storage"
	"github.com/khulnasoft/codeinfra/sdk/v3/go/codeinfra"
	"github.com/khulnasoft/codeinfra/sdk/v3/go/codeinfra/config"
)

func main() {
	codeinfra.Run(func(ctx *codeinfra.Context) error {
		cfg := config.New(ctx, "")
		// The name of the storage account
		storageAccountNameParam := cfg.Require("storageAccountNameParam")
		// The name of the resource group
		resourceGroupNameParam := cfg.Require("resourceGroupNameParam")
		resourceGroupVar, err := core.LookupResourceGroup(ctx, &core.LookupResourceGroupArgs{
			Name: resourceGroupNameParam,
		}, nil)
		if err != nil {
			return err
		}
		locationParam := resourceGroupVar.Location
		if param := cfg.Get("locationParam"); param != "" {
			locationParam = param
		}
		storageAccountTierParam := "Standard"
		if param := cfg.Get("storageAccountTierParam"); param != "" {
			storageAccountTierParam = param
		}
		storageAccountTypeReplicationParam := "LRS"
		if param := cfg.Get("storageAccountTypeReplicationParam"); param != "" {
			storageAccountTypeReplicationParam = param
		}
		storageAccountResource, err := storage.NewAccount(ctx, "storageAccountResource", &storage.AccountArgs{
			Name:                   codeinfra.String(storageAccountNameParam),
			AccountKind:            codeinfra.String("StorageV2"),
			Location:               codeinfra.String(locationParam),
			ResourceGroupName:      codeinfra.String(resourceGroupNameParam),
			AccountTier:            codeinfra.String(storageAccountTierParam),
			AccountReplicationType: codeinfra.String(storageAccountTypeReplicationParam),
		})
		if err != nil {
			return err
		}
		ctx.Export("storageAccountNameOut", storageAccountResource.Name)
		return nil
	})
}
